deploy/supervisor/deployment.yaml

```diff
@@ -115,8 +115,6 @@ spec:
               readOnly: false  #! writable to allow for socket use
             #@ end
           ports:
-            - containerPort: 8080
-              protocol: TCP
```

Comment:
The Kubernetes docs for `ports` says:

> List of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational. Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network. Cannot be updated.

So it shouldn't matter if it is listed here.

However, now that I look at that doc again it also says "Cannot be updated" so I wonder if changing this will break upgrades.

Reply:
It seems like deploying with both ports listed and then deploying again with only one port listed to update the deployment works on a Kind 1.23 cluster. Any idea what they mean by "Cannot be updated"?

Reply:
> It seems like deploying with both ports listed and then deploying again with only one port listed to update the deployment works on a Kind 1.23 cluster. Any idea what they mean by "Cannot be updated"?

It is likely referring to an in-place update of the pod not allowing this field to change.  On upgrade, the deployment controller will rollout new pods since the pod template hash would have changed.